components/icon/badge-icon/index.jsx

```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// ### React
+import React from 'react';
+import PropTypes from 'prop-types';
+
+// ### classNames
+// [github.com/JedWatson/classnames](https://github.com/JedWatson/classnames)
+// This project uses `classnames`, "a simple javascript utility for conditionally
+// joining classNames together."
+import classNames from 'classnames';
+
+// This component's `checkProps` which issues warnings to developers about properties when in development mode (similar to React's built in development tools)
+import checkProps from './check-props';
+
+// ### Children
+import SLDSUtilityIcon from '../../utilities/utility-icon';
+
+import { BADGE_ICON } from '../../../utilities/constants';
+
+/**
+ * This is a non-interactive wrapper component for `UtilityIcon` that specifies badge icon classes for an icon inside a `badge` tag. Use of this component by itself is not recommended, but should be used as part of other components to obtain the correct styling for icons within badges. It does not return a `badge`. It only returns an icon for use within a badge. Assistive text must also be rendered by the parent.
+ */
+const BadgeIcon = (props) => {
+	checkProps(BADGE_ICON, props);
+
+	return (
+		<span className={`slds-badge__icon slds-badge__icon_${props.position}`}>
+			<span
+				className={`slds-icon_container slds-icon-${props.category}-${
```

Comment:
Look at [here](https://github.com/salesforce/design-system-react/blob/master/docs/codebase-overview.md#use-entire-class-name-string-when-rendering-conditional-class-names) on the coding standard for writing class names :)